container-host-files/etc/hcf/config/scripts/fetch_uaa_verification_key.sh

```diff
@@ -7,19 +7,50 @@
 
 # Note that this is *sourced* into run.sh, so we can't exit the shell.
 
-# Wait for UAA
-while true ; do
-    if curl --fail \
-        $(if test "${SKIP_CERT_VERIFY_EXTERNAL}" = "true" ; then echo "--insecure" ; fi) \
-        "${HCF_UAA_INTERNAL_URL}/token_key"
-    then
-        break
-    fi
-    sleep 10
-done
+
+# Report progress to the user; use as printf
+status() {
+    local fmt="${1}"
+    shift
+    printf "\n%b${fmt}%b\n" "\033[0;32m" "$@" "\033[0m"
+}
+
+# Report problem to the user; use as printf
+trouble() {
+    local fmt="${1}"
+    shift
+    printf "\n%b${fmt}%b\n" "\033[0;31m" "$@" "\033[0m"
+}
+
+# helper function to retry a command several times, with a delay between trials
+# usage: retry <max-tries> <delay> <command>...
+function retry_forever () {
+    delay=${1}
+    shift 1
+
+    while true ; do
+	printf "Trying: %s\n" "$*"
```

Comment:
I wanted the log to show something for every iteration, yes.
So that we can see the role is still doing something.
Printing the command was not only the easiest, but also the surest way. As the command itself my not produce output.
There also is the issue of the command scrolling of screen if we have enough iterations and thus making it more difficult to see the url. Especially if we start looking at the log late when role is not coming up. (The available delay may not be enough to scroll far enough back before the new output snaps it back to the bottom).